src/gui/lineedit/src/lineedit.h

```diff
@@ -12,26 +12,29 @@
 
 #include <QLineEdit>
 
-QT_BEGIN_NAMESPACE
 class QToolButton;
-QT_END_NAMESPACE
 
 class LineEdit : public QLineEdit
 {
   Q_OBJECT
 
 public:
-  LineEdit(QWidget *parent = 0);
```

Comment:
> QWidget without parent it's correct use case

I think you mean valid use case, then yes.

>OOP is not intended to protect the programmer from wrong actions (it's impossible).

It's not about OOP here, it's just the old (bad) code in #3858 kinda frustrate me, that PR shouldn't exist if the `*parent` were passed correctly
That's why I wanted to remove default value, you can still pass `0` if the programmer knows what he/she is doing anyway.


I'll change it back if no further discussion later.